src/common/health.rs

```diff
@@ -128,14 +128,20 @@ impl Task {
         while self.commit_index() < cluster_commit_index {
             // If not:
             //
-            // - Refresh cluster commit index
-            let Some(current_cluster_commit_index) = self.cluster_commit_index().await else {
+            // - Check if this is the only node in the cluster
+            let Some(_) = self.cluster_commit_index().await else {
                 self.set_ready();
                 return;
             };
 
-            // - Check if cluster commit index *decresed* since last check
-            cluster_commit_index = cmp::min(current_cluster_commit_index, cluster_commit_index);
+            // TODO: Do we want to update `cluster_commit_index` here?
+            //
```

Comment:
We also did this `min` of the `cluster_commit_index` on each iteration, but Raft guarantees that it will never decrease cluster commit index.

And so, if during any of this continuous queries, *all* requests fail (e.g., because node lost connectivity for a moment) then `cluster_commit_index()` will return 0, and `min(cluster_commit_index, 0)` *is* 0. So we will instantly go to the next step of the `/readyz` check, even though we have *some* reasonable commit already.

Basically, I think this also makes no sense and makes `/readyz` check more flaky, so I've removed the `min` call.